### ISSUE TITLE
Reactor rule php8 ReplaceNewDateTimeNull has wrong implementation

### ISSUE THREAD

**Issue by 0m3r** [magento/magento-coding-standard#392]
### Description
I have a couple of wrong passes of the rule 
```diff

-        $date = new \DateTime($date);
-        $now = new \DateTime();
+        $date = new \DateTime('now');
+        $now = new \DateTime('now');
         $now->setTime(0, 0, 0, 0);
         $minDelay = $this->getMinDelayDays();
         $interval = $now->diff($date);
    ----------- end diff -----------

Applied rules:
 * ReplaceNewDateTimeNull
```

```data_c.php
public function __construct(
        #[LanguageLevelTypeAware(['8.0' => 'string'], default: '')] $datetime = 'now',
        #[LanguageLevelTypeAware(['8.0' => 'DateTimeZone|null'], default: 'DateTimeZone')] $timezone = null
    ) {}
```
default is 'now' and it's a string so that the first argument can be empty and it's also can be variable  
but in rule it's always must by string

https://github.com/magento/magento-coding-standard/blob/50e7c2ad3d7110b06527d75a1aa954b7c20f6ea1/Magento2/Rector/Src/ReplaceNewDateTimeNull.php#L36
```php
if ($node->args[0] !== String_::class) {
```

### Expected behavior
it looks like this rule should  work only if the first argument is precisely null
